Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -1104,38 +1104,26 @@ int CLuaEngineDefs::EngineGetVisibleTextureNames(lua_State* luaVM)
     return 1;
 }
 
-int CLuaEngineDefs::EngineSetModelVisibleTime(lua_State* luaVM)
+bool CLuaEngineDefs::EngineSetModelVisibleTime(const std::variant<std::string, unsigned int> variantModelID, char hourOn, char hourOff)
 {
-    // bool engineSetModelVisibleTime ( int/string modelID, int hourOn, int hourOff )
-    SString strModelId;
-    char cHourOn,cHourOff;
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadString(strModelId);
-    argStream.ReadNumber(cHourOn);
-    argStream.ReadNumber(cHourOff);
+    if (hourOn > 24 || hourOn < 0)
+        throw std::invalid_argument("hour on must be between 0 and 24");
```

Comment:
EngineSetModelVisibleTime

```suggestion
        throw std::invalid_argument("hourOn must be between 0 and 24");
```